### PR TITLE
Load, reload and display Locations on the map

### PR DESCRIPTION
**Pull request by Tazaf**
Create an ApiService that manages the communication with the API backend.
All other services that need to request the BioPocket API should include
this ApiService.

Create an EnvService that abstract the use of the `ENV` variable. An
EnvInterface is also created, and must be updated each time the `ENV`
variable needs to be updated.
You should include the EnvService wherever you need to access the `ENV`
variable. This helps a lot when creating test as the EnvService can be
easily mocked, which is not the case of the `ENV` variable.

Beware that the `environment.sample.ts` file has changed !
You should update all your `environment.*.ts` files to match this new
sample.

Install `ng-request-builder` to easily make Http requests.
Note that this package will probably be removed when Angular will be
updated to its version 5.*

All services and classes that only export one thing now uses the `default`
keyword, so that we can import them without using destructured
assignments.

The `copy.config.js` file is used to copy assets, stylesheets or scripts
from node modules into the `www` directory, so that they can be referenced
and used at runtime.
See `DEVELOPMENT.md` for more information.

The `Print` utility should be used instead of `console` when you need some
permanent-ish logs, as it does only print logs when allowed to do so.

Story: TG-43

### CODE REVIEW

File: src/providers/locations-service/locations-service.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Handles request on the BioPocket API that are related to the management of Locations of interest 
+ */
+@Injectable()
+export default class LocationsService {
+
+  private resourceName: string = '/locations'
+
+  constructor(private api: ApiService) {
+    this.api.version().subscribe(ver => Print.log(ver));
```
Comment (AlphaHydrae) [inline src/providers/locations-service/locations-service.ts]: This is not a good place to put this call. We might copy-paste it by mistake into other services. `AppComponent` is probably a better place.

File: src/utils/leafletIcons.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { icon as IconFactory, Icon, IconOptions } from 'leaflet';
+
+/**
+ * The default Icon used by Leaflet.
+ * The iconUrl, iconRetinaUrl and shadowUrl have been overwritten so that the images are properly retrieved.
+ * 
+ * The icon is created with an IIFE that creates a IconDefault instance, updates the required options and return
+ * a new Icon with these default updated options.
+ * This way, we don't have to manually redefined all options for the default marker.
```
Comment (AlphaHydrae) [inline src/utils/leafletIcons.ts]: `redefined` -> `redefine`

**Comment by AlphaHydrae** [comment]
There's a bunch of lint warnings that we should fix. It's only "unused" warnings for now, but we might miss something important someday if we ignore them all the time.

The warnings are displayed on the command line when running `ionic serve` (but only once at start time), or when running `npm run lint`.

```
[11:15:19]  tslint: src/providers/api-service/api-service.ts, line: 8 
            All imports are unused. 

       L7:  import { ApiVersion } from '../../models';
       L8:  import Print from '../../utils/print';

[11:15:19]  tslint: src/app/app.component.spec.ts, line: 28 
            All imports are unused. 

      L27:  import LocationsModule from '../providers/locations-service/locations-module';
      L28:  import LocationsService from '../providers/locations-service/locations-service';
      L29:  import { translateModuleForRoot } from '../utils/i18n';

[11:15:19]  tslint: src/pages/map/map.spec.ts, line: 22 
            'lastConnection' is declared but never used. 

      L21:  describe('MapPage', function () {
      L22:    let component, fixture, backend, lastConnection;
      L23:    let navControllerMock;

[11:15:19]  tslint: src/pages/map/map.spec.ts, line: 24 
            'responses' is declared but never used. 

      L23:    let navControllerMock;
      L24:    let responses;

[11:15:19]  tslint: src/providers/api-service/api-service.spec.ts, line: 4 
            'inject' is declared but never used. 

       L4:  import { TestBed, fakeAsync, inject, tick } from '@angular/core/testing';
       L5:  import { HttpModule, Http, ConnectionBackend, RequestOptions, BaseRequestOptions, Response, ResponseOptions } from '@angular/http';

[11:15:19]  tslint: src/providers/api-service/api-service.spec.ts, line: 5 
            'RequestOptions' is declared but never used. 

       L4:  import { TestBed, fakeAsync, inject, tick } from '@angular/core/testing';
       L5:  import { HttpModule, Http, ConnectionBackend, RequestOptions, BaseRequestOptions, Response, ResponseOptions 
       L6:  import { MockBackend } from '@angular/http/testing';

[11:15:19]  tslint: src/providers/api-service/api-service.spec.ts, line: 5 
            'BaseRequestOptions' is declared but never used. 

       L4:  import { TestBed, fakeAsync, inject, tick } from '@angular/core/testing';
       L5:  import { HttpModule, Http, ConnectionBackend, RequestOptions, BaseRequestOptions, Response, ResponseOptions 
       L6:  import { MockBackend } from '@angular/http/testing';

[11:15:19]  lint finished in 2.93 s
```

File: src/providers/locations-service/locations-service.ts (hunk 1 above)
Comment (Tazaf) [reply src/providers/locations-service/locations-service.ts]: Yep, I forgot to remove it.
This was only used for the tests.
There's indeed no need for this to be there.

File: src/models/base-marker.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { MarkerOptions, LatLngExpression, Marker } from 'leaflet';
+
+interface BaseMarkerData {
+}
+
+export default class BaseMarker extends Marker {
```
[the remark this answers is not in the document]
Comment (Tazaf) [reply src/models/base-marker.ts]: I like this idea.
Will do.

**Pull request merged**